library/src/main/java/com/owncloud/android/lib/resources/files/SearchRemoteOperation.java

```diff
@@ -169,5 +175,82 @@ protected RemoteOperationResult<List<RemoteFile>> run(OwnCloudClient client) {
         }
         return result;
     }
+
+    @Override
+    public RemoteOperationResult run(NextcloudClient client) {
+        RemoteOperationResult<List<RemoteFile>> result;
+        NcSearchMethod searchMethod = null;
+
+        String webDavUrl = client.getDavUri().toString();
+
+        SearchInfo searchInfo = new SearchInfo("NC",
+                Namespace.XMLNS_NAMESPACE,
+                searchQuery);
+
+        try {
+            searchMethod = new NcSearchMethod(webDavUrl,
+                    searchInfo,
+                    searchType,
+                    getClientNew().getUserIdPlain(),
+                    timestamp,
+                    limit,
+                    filterOutFiles,
+                    capability,
+                    startDate,
+                    endDate);
+
+            // disable redirect
+            OkHttpClient disabledRedirectClient = client.getClient()
+                    .newBuilder()
+                    .followRedirects(false)
+                    .authenticator(new NextcloudAuthenticator(client.getCredentials(), "Authorization"))
+                    .build();
```

Comment:
I don't think it's a good idea to create a new OkHttpClient for every request...  the `NextcloudClient` passed to `run` should already be authenticated IMO. Not sure about the redirects or why they are disabled.